src/snipster/db.py

```diff
@@ -10,14 +11,41 @@
 engine = create_engine(db_url, echo=False)
 
 
-@contextmanager
-def get_session():
-    """Yield a database session, properly closing it after use."""
-    session = Session(engine)
-    try:
-        yield session
-    except Exception:
-        session.rollback()
-        raise
-    finally:
-        session.close()
+class SessionFactory:
+    def __init__(self, engine):
+        self.engine = engine
+        self._sessions = []  # Track sessions
```

Comment:
I think each session is already scoped within its context manager, why would we track multiple sessions?